File: app/src/main/java/org/gittner/osmbugs/statics/Images.java
package org.gittner.osmbugs.statics;

import android.content.Context;
import android.content.res.Resources;
import android.graphics.drawable.Drawable;
import android.support.v4.content.ContextCompat;

public class Images
{
    private static Context mContext;
    private static Resources mResources;
    private static String mPackageName;


    public static void init(Context context)
    {
        mContext = context;
        mResources = context.getResources();
        mPackageName = context.getPackageName();
    }


    public static Drawable getByName(String name, int defaultDrawableId)
    {
        int id = mResources.getIdentifier(name, "drawable", mPackageName);

        try
        {
            return ContextCompat.getDrawable(mContext, id);
        }
        catch (android.content.res.Resources.NotFoundException e)
        {
            return get(defaultDrawableId);
        }
    }


    public static Drawable get(int id)
    {
        return ContextCompat.getDrawable(mContext, id);
    }
}

File: app/src/main/java/org/gittner/osmbugs/statics/Images.kt
package org.gittner.osmbugs.statics

import android.content.res.Resources
import android.graphics.drawable.Drawable
import androidx.core.content.res.ResourcesCompat
import org.gittner.osmbugs.App
import org.gittner.osmbugs.keepright.KeeprightError
import org.gittner.osmbugs.mapdust.MapdustError
import org.gittner.osmbugs.osmnotes.OsmNote
import org.gittner.osmbugs.osmose.OsmoseError

class Images {
    companion object {
        private lateinit var mResources: Resources

        /**
         * Returns a Drawable for the given id
         * @param id The ID of the Drawable
         */
        fun GetDrawable(id: Int): Drawable {
            return ResourcesCompat.getDrawable(mResources, id, null)!!
        }

        fun init(app: App) {
            mResources = app.applicationContext.resources

            OsmNote.Init()
            KeeprightError.Init()
            MapdustError.Init()
            OsmoseError.Init()
        }
    }
}